blog/_posts/2024-06-06-istio-native-sidecar.md

```diff
@@ -0,0 +1,164 @@
+---
+title: "Introducing Native Sidecar Support for Istio addon on AKS"
+description: "Explore native sidecar support for the Istio add-on in Azure Kubernetes Service (AKS). Learn its benefits, how to enable it, and see a practical demo."
+date: 2024-06-06
+author: Fuyuan Bie
+categories:
+  - networking
+  - addons
+---
+
+We're excited to announce the public preview of native sidecar support for Istio add-on for Azure Kubernetes Service (AKS). This blog post will guide you through the fundamentals of Kubernetes native sidecars, the integration of native sidecar support in Istio add-on, the benefits of this feature, how to enable it, and a demo showcasing its advantages.
+
+## Understanding Kubernetes Native Sidecars
+
+In Kubernetes, the sidecar pattern is used to extend the functionality of applications running in pods. Sidecar containers are responsible for auxiliary tasks such as logging, proxying, and monitoring. Typically, managing sidecar lifecycle required special configurations, which could be complex and error-prone. Developers must understand the lifecycle interactions of containers within a Kubernetes pod to avoid unexpected behaviors. They cannot safely assume that sidecars will be ready before their application or expect a specific shutdown order. Additionally, for jobs, the primary job container must terminate the sidecars.
+
+Starting from Kubernetes v1.28, native sidecar support becomes a builtin feature. On AKS, starting from v1.29, this feature is turned on.
+
+Read more about this feature from [Kubernetes blog](https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/).
+
+## Native Sidecar Support in Istio 1.20+
+
+With the release of Istio 1.20, native sidecar support has been introduced, offering a streamlined approach to managing sidecar proxies. This integration reduces the overhead and complexity associated with sidecar injection, providing efficient management of sidecar life cycles.
+
+Read more about Istio native sidecar support from [Istio blog](https://istio.io/latest/blog/2023/native-sidecars/).
+
+## Where Native Sidecar Does not Help
+
+While native sidecar support addresses many challenges, it does not assist with egress traffic as much as with ingress traffic. This is because within the lifecycle of a pod, application containers are terminated after sidecar containers, leaving no effective way to notify the application container that the pod is being terminated before the native sidecar is stopped. Consequently, in-flight egress traffic from the application container may experience disruptions during pod restarts.
+
+## Try it out!
+
+### Enabling Istio Native Sidecar Support in AKS
+
+Since this is currently a preview feature, we need to follow [these instructions](https://learn.microsoft.com/en-us/azure/aks/istio-native-sidecar#before-you-begin) to register `IstioNativeSidecarModePreview` preview feature.
```

Comment:
Removing locale from MS learn URL.

```suggestion
Since this is currently a preview feature, we need to follow [these instructions](https://learn.microsoft.com/azure/aks/istio-native-sidecar#before-you-begin) to register `IstioNativeSidecarModePreview` preview feature.
```